kratos/factories/standard_builder_and_solver_factory.cpp

```diff
@@ -0,0 +1,51 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "factories/base_factory.h"
+#include "spaces/ublas_space.h"
+
+// Builder And Solver
+#include "solving_strategies/builder_and_solvers/builder_and_solver.h"
+#include "solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h"
+#include "solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h"
+#include "solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h"
+
+namespace Kratos
+{
+    void RegisterBuilderAndSolvers()
+    {
+        typedef TUblasSparseSpace<double> SpaceType;
+        typedef TUblasDenseSpace<double> LocalSpaceType;
+        typedef LinearSolver<SpaceType, LocalSpaceType> LinearSolverType;
+
+        typedef ResidualBasedEliminationBuilderAndSolver<SpaceType,  LocalSpaceType, LinearSolverType> ResidualBasedEliminationBuilderAndSolverType;
+        typedef ResidualBasedEliminationBuilderAndSolverWithConstraints<SpaceType,  LocalSpaceType, LinearSolverType> ResidualBasedEliminationBuilderAndSolverWithConstraintsType;
+        typedef ResidualBasedBlockBuilderAndSolver<SpaceType,  LocalSpaceType, LinearSolverType> ResidualBasedBlockBuilderAndSolverType;
+
+        //NOTE: here we must create persisting objects for the builder and solvers
+        static ResidualBasedEliminationBuilderAndSolverType msResidualBasedEliminationBuilderAndSolver;
```

Comment:
@RiccardoRossi here you can see like I declare static empty B&S, so that's why I need to check that the B&S linear solver pointer is null